src/libs/metadata.c

```diff
@@ -475,21 +476,30 @@ static void config_button_clicked(GtkButton *button, dt_lib_module_t *self)
 
   if (gtk_dialog_run(GTK_DIALOG(dialog)) == GTK_RESPONSE_YES)
   {
+    gboolean meta_signal = FALSE;
+    gboolean meta_remove = FALSE;
     for(unsigned int i = 0; i < DT_METADATA_NUMBER; i++)
     {
       const int type = dt_metadata_get_type_by_display_order(i);
       if(type != DT_METADATA_TYPE_INTERNAL)
       {
         char *setting = dt_util_dstrcat(NULL, "plugins/lighttable/metadata/%s_hidden", name[i]);
         const gboolean hidden_nv = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(hidden[i]));
-        if(hidden_nv !=  hidden_v[i]) dt_conf_set_bool(setting, hidden_nv);
+        if(hidden_nv !=  hidden_v[i])
+        {
+          dt_conf_set_bool(setting, hidden_nv);
+          meta_signal = TRUE;
+          meta_remove =  hidden_nv ? TRUE : meta_remove;
+        }
         g_free(setting);
         setting = dt_util_dstrcat(NULL, "plugins/lighttable/metadata/%s_private", name[i]);
         const gboolean private_nv = gtk_toggle_button_get_active(GTK_TOGGLE_BUTTON(private[i]));
         if(private_nv != private_v[i]) dt_conf_set_bool(setting, private_nv);
         g_free(setting);
       }
     }
+    if(meta_signal)
+      dt_control_signal_raise(darktable.signals, DT_SIGNAL_METADATA_CHANGED, meta_remove ? 1 : 0);
```

Comment:
and 0 or 1 here.

Those values seems to control something, I would certainly prefer having an enum with proper names.

Reply:
Yes, sure !